panel/config/configpaneldialog.cpp

```diff
@@ -42,9 +42,9 @@ ConfigPanelDialog::ConfigPanelDialog(LXQtPanel *panel, QWidget *parent):
     addPage(mPluginsPage, tr("Widgets"), QLatin1String("preferences-plugin"));
     connect(this, &ConfigPanelDialog::reset, mPluginsPage, &ConfigPluginsWidget::reset);
 
-    connect(this, &ConfigPanelDialog::accepted, [panel] {
+    connect(this, &ConfigPanelDialog::accepted, this, [panel] {
         panel->saveSettings();
-   });
+    });
```

Comment:
Oh, I missed this one:

The other instances of `this` you've added as lambda contexts aren't really needed, although they're harmless. However, in the above connection, the missing context is `panel`, not `this`. Please correct it!

Clazy may show warnings about lambda contexts (I don't use Clazy) but, if so, that's a problem in clazy: it isn't smart enough to know about them.

Reply:
Clazy is just a tool for static code analysis. I'm adding the contexts because of [this](https://github.com/KDE/clazy/blob/master/docs/checks/README-connect-3arg-lambda.md).
Forgive me but I just can't see why you think the missing context there is "panel". The lambda captures the panel object and just call it (panel->saveSettings()).

Reply:
> The lambda captures the panel object and just call it (panel->saveSettings()).

And with `panel` as the context, we make sure that `panel` exists when its method is called -- no crash.

Reply:
Right :)